core/src/networked_client.cpp

```diff
@@ -0,0 +1,219 @@
+                transform.set_position(serMess->playerPos[playerNum]);
+                transform.set_rotation(serMess->playerRots[playerNum]);
+                rigidBody.set_transform(serMess->playerPos[playerNum], serMess->playerRots[playerNum]);
+                // TODO: set player health, pass the paint collisions to somewhere they can be handled?
+            }
+        }
+        m_client->ReleaseMessage(mess);
+    }
+    return true;
+}
+bool NetworkedClient::send_message() {
+    if (m_tickTime >= TICK) {
+        if (m_client->CanSendMessage(1)) {
+            Message* mess = generate_message();
+            if (mess) {
+                m_client->SendMessage(1, mess);
+                seq++;
+            }
+        } else {
+            return false;
+        }
+    }
+    return true;
+}
+Message* NetworkedClient::generate_message() {
+    auto sceneOpt = Scene::get_active_scene();
+    if (!sceneOpt) {
+        return nullptr;
+    }
+    Scene& scene = sceneOpt.value();
+    entt::registry& registry = scene.get_registry();
+
+    auto players = registry.view<components::ClientPlayer>();
+
+    for (auto playerHandle : players) {
+        auto playerOpt = scene.get_game_object_from_handle(playerHandle);
+        if (!playerOpt) {
+            continue;
+        }
+        auto playerGO = playerOpt.value();
+        auto& playerComp = playerGO.get_component<components::ClientPlayer>();
+
+        if (!(playerComp.get_client_num() == m_clientNum)) {
+            continue;
+        }
+        auto* mess = (ClientMessage*)m_client->CreateMessage(MessageTypes::CLIENT_MESSAGE);
+        mess->set_sequence(seq);
+        mess->set_ack(serverSeq);
+        mess->m_lookAxis = playerComp.get_look_axis();
+        mess->m_moveAxis = playerComp.get_movement_axis();
+        mess->jumpPressed = playerComp.get_jumping_pressed();
+        mess->isShooting = playerComp.get_is_shooting();
+        return mess;
+    }
+    return nullptr;
+}
+void NetworkedClient::test_player_input() {
+    auto sceneOpt = Scene::get_active_scene();
+    if (!sceneOpt) {
+        return;
+    }
+    Scene& scene = sceneOpt.value();
+    entt::registry& registry = scene.get_registry();
+
+    auto players = registry.view<components::ClientPlayer>();
+
+    for (auto playerHandle : players) {
+        auto playerOpt = scene.get_game_object_from_handle(playerHandle);
+        if (!playerOpt) {
+            continue;
+        }
+        auto playerGO = playerOpt.value();
+        auto& playerComp = playerGO.get_component<components::ClientPlayer>();
+
+        if (!(playerComp.get_client_num() == m_clientNum)) {
+            continue;
+        }
+
+        playerComp.set_is_shooting(!playerComp.get_is_shooting());
+        playerComp.set_jumping_pressed(!playerComp.get_jumping_pressed());
+
+        vec3 m_playerInputs = vec3();
+        if (m_inManager->key_pressed(KeyCode::A)) {
+            m_playerInputs.x -= 1;
```

Comment:
framerate dependent code consider passing it in as a param to the function

Reply:
This shouldn't be as it only handles input? not the actual movement of players?

Reply:
oke